java.lang.IllegalStateException: InteliJ debug agent detected, disable it to prevent unexpected instrumentation errors

Even though https://github.com/elastic/apm-agent-java/issues/1673 got fixed a while ago in the latest IntelliJ IDEA 2023.2.1 (Ultimate Edition) with latest agent 1.42.0 we can see : 

```
[elastic-apm-agent] ERROR Failed to start agent
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at co.elastic.apm.agent.premain.AgentMain.loadAndInitializeAgent(AgentMain.java:157)
	at co.elastic.apm.agent.premain.AgentMain.init(AgentMain.java:98)
	at co.elastic.apm.agent.premain.AgentMain.agentmain(AgentMain.java:58)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:491)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallAgentmain(InstrumentationImpl.java:513)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at co.elastic.apm.agent.util.DependencyInjectingServiceLoader.instantiate(DependencyInjectingServiceLoader.java:129)
	at co.elastic.apm.agent.util.DependencyInjectingServiceLoader.instantiate(DependencyInjectingServiceLoader.java:119)
	at co.elastic.apm.agent.util.DependencyInjectingServiceLoader.<init>(DependencyInjectingServiceLoader.java:70)
	at co.elastic.apm.agent.util.DependencyInjectingServiceLoader.load(DependencyInjectingServiceLoader.java:90)
	at co.elastic.apm.agent.bci.ElasticApmAgent.loadInstrumentations(ElasticApmAgent.java:188)
	at co.elastic.apm.agent.bci.ElasticApmAgent.initInstrumentation(ElasticApmAgent.java:180)
	at co.elastic.apm.agent.bci.ElasticApmAgent.initialize(ElasticApmAgent.java:166)
	... 13 more
Caused by: java.lang.IllegalStateException: IntelliJ debug agent detected, disable it to prevent unexpected instrumentation errors. See https://github.com/elastic/apm-agent-java/issues/1673
	at co.elastic.apm.agent.concurrent.ForkJoinTaskInstrumentation.<clinit>(ForkJoinTaskInstrumentation.java:52)
	... 22 more
```


## Describe the bug
Intellij: 2023.2.1
Elastic APM agent: 1.42.0
JDK: openjdk 17.0.8 or 20.0.2

Instrumentation doesn't work in debug mode in Intellij IDE. 

## Expected behavior
Elastic APM agent can start without errors

--- a/CHANGELOG.asciidoc
+++ b/CHANGELOG.asciidoc
@@ -44,6 +44,7 @@ Use subheadings with the "=====" level for adding notes for unreleased changes:
 * Added `baggage_to_attach` config option to allow automatic lifting of baggage into transaction, span and error attributes - {pull}3288[#3288], {pull}3289[#3289]
 * Exclude elasticsearch 8.10 and newer clients from instrumentation because they natively support OpenTelemetry  - {pull}3303[#3303]
 * Switched to OpenTelemetry compatible context propagation for Kafka - {pull}3300[#3300]
+* Allow running the IntelliJ debug agent in parallel - {pull}3315[#3315]
 
 [[release-notes-1.x]]
 === Java Agent version 1.x

--- a/apm-agent-plugins/apm-java-concurrent-plugin/src/main/java/co/elastic/apm/agent/concurrent/ForkJoinTaskInstrumentation.java
+++ b/apm-agent-plugins/apm-java-concurrent-plugin/src/main/java/co/elastic/apm/agent/concurrent/ForkJoinTaskInstrumentation.java
@@ -42,17 +42,6 @@ public class ForkJoinTaskInstrumentation extends ElasticApmInstrumentation {
 
     private static final Tracer tracer = GlobalTracer.get();
 
-    static {
-        if (Boolean.parseBoolean(System.getProperty("intellij.debug.agent"))) {
-            // IntelliJ debugger also instrument some java.util.concurrent classes and changes the class structure.
-            // However, the changes are not re-applied when re-transforming already loaded classes, which makes our
-            // agent unable to see those structural changes and try to load classes with their original bytecode
-            //
-            // Go to the following to enable/disable: File | Settings | Build, Execution, Deployment | Debugger | Async Stack Traces
-            throw new IllegalStateException("IntelliJ debug agent detected, disable it to prevent unexpected instrumentation errors. See https://github.com/elastic/apm-agent-java/issues/1673");
-        }
-    }
-
     @Override
     public ElementMatcher<? super TypeDescription> getTypeMatcher() {
         return is(ForkJoinTask.class);

